src/txmempool.h

```diff
@@ -338,7 +338,7 @@ class CTxMemPool
      * check does nothing.
      */
     void check(const CCoinsViewCache *pcoins) const;
-    void setSanityCheck(bool _fSanityCheck) { fSanityCheck = _fSanityCheck; }
+    void setSanityCheck(double dFrequency = 1.0) { nCheckFrequency = dFrequency * 4294967296.0; }
```

Comment:
Also, ULONG_MAX gives the maximum value of an unsigned long. Here we need
the number of possible values in a uint32_t. That is by definition 2^32.